Add formatters and apply rules; code refactoring

pat-s opened the pull request:
@vitabaks Lots of changes but in the end they will result in a consistent formatting for all files. 

Due to the amount of changes I went for a dedicated PR. Sometimes changes to quotes might result in changed behaviour. Not sure if the tests cover everything here, let's see how it goes.

You could optionally add some CI checks to enforce these linters in PRs.

## Config files

- `.prettierrc.json`
- `.editorconfig`
- `.sql-formatter.json`

## Commands

- `prettier -w .`
- `editorconfig-checker .`

## Makefile

`make help`
```shell
## —— Formatting ——————————————————————————————————————————————————————————————————————————-------  
prettier                       Run Prettier formatting 
prettier-check                 Check formatting with Prettier (without modifying files) 
sql-format                     Format all SQL files using sql-formatter 
pg-format                      Format all SQL files using pgFormatter (PostgreSQL SQL queries and PL/PGSQL code beautifier) 
```

vitabaks commented:
Indeed the change is quite big.

Should we stick to these parameters for the entire project or just for the [automation](https://github.com/vitabaks/autobase/tree/master/automation) component (ansible)?  I'm concerned that this might break the Console UI/API, which I didn't develop myself—it's custom-built by a third-party company. If something breaks, I'll have to cover the cost of refactoring. 🙂

vitabaks commented:
@pat-s Please pay attention to linter errors

pat-s commented:
> Should we stick to these parameters for the entire project or just for the [automation](https://github.com/vitabaks/autobase/tree/master/automation?rgh-link-date=2025-03-05T10%3A38%3A05Z) component (ansible)?

Can also do that, I personally care most/only for this part 😄️

Will clean up linters and only apply changes to `automation/` then.

pat-s commented:
- Reverted `console/` changes
- Adjusted linter configs to work with prettier
- Fixed various `ansible-lint` reports
- Updated `requirements.yml`
- Avoid using reserved keyword `query` as var name

vitabaks commented on automation/config_pgcluster.yml:
```diff
@@ -9,15 +9,15 @@
     # Note: Applicable only for "aws", "gcp", "azure", because:
-      # "digitalocean" - requires the Spaces access keys "AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY"
-      # "hetzner" - currently, Hetzner Cloud does not provide S3 storage
+    # "digitalocean" - requires the Spaces access keys "AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY"
+    # "hetzner" - currently, Hetzner Cloud does not provide S3 storage
```
```suggestion
    # "digitalocean" - requires the Spaces access keys ("AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY" variables)
    # "hetzner" - requires the S3 credentials ("hetzner_object_storage_access_key" and "hetzner_object_storage_secret_key" variables).
```

vitabaks commented on automation/deploy_pgcluster.yml:
```diff
@@ -9,15 +9,15 @@
     # Note: Applicable only for "aws", "gcp", "azure", because:
-      # "digitalocean" - requires the Spaces access keys "AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY"
-      # "hetzner" - currently, Hetzner Cloud does not provide S3 storage
+    # "digitalocean" - requires the Spaces access keys "AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY"
+    # "hetzner" - currently, Hetzner Cloud does not provide S3 storage
```
```suggestion
    # "digitalocean" - requires the Spaces access keys ("AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY" variables)
    # "hetzner" - requires the S3 credentials ("hetzner_object_storage_access_key" and "hetzner_object_storage_secret_key" variables).
```

vitabaks commented on automation/requirements.yml:
```diff
@@ -10,3 +10,7 @@ collections:
     version: "==1.27.0"
   - name: hetzner.hcloud
     version: "==4.2.2"
+  - name: ansible.windows
+    version: "==2.7.0"
+  - name: community.postgresql
+    version: "==3.10.2"
```
```suggestion
```
This is not a requirement for this project.

pat-s replied:
Is is in `patroni/tasks` and `consul/handlers`? Besides, I am getting errors for `ansible-lint` if these are not installed.

vitabaks replied:
`community.postgresql` is possible, but windows is unlikely to be supported)

vitabaks replied:
>  am getting errors for ansible-lint if these are not installed.

ok

vitabaks replied:
> am getting errors for ansible-lint if these are not installed.
we could add an exception to the linter instead of installing a windows component that we don't actually need.

pat-s replied:
It's in the module, so 🤷️ Don't wanna discuss about it, this is your expertise area. Just stating that these calls exists, which is why I've added them.

vitabaks commented on automation/roles/authorized-keys/tasks/main.yml:
```diff
@@ -12,7 +11,7 @@
         key: "{{ item }}"
         state: present
-      loop: "{{ ssh_public_keys_list | map('replace', '\"', '') | map('replace', \"'\", \"\") | list }}"
+      loop: '{{ ssh_public_keys_list | map(''replace'', ''"'', '''') | map(''replace'', "''", "") | list }}'
```
```suggestion
      loop: "{{ ssh_public_keys_list | map('replace', '\"', '') | map('replace', \"'\", \"\") | list }}"
```
The change doesn’t bring significant improvement as it replaces escaping double quotes with escaping single quotes. Since we are standardizing on double quotes, it’s better to keep the original format.

@pat-s what do you think?

pat-s replied:
These were automated changes, you can change/adapt any as you like. The linter will complain, if there's an issue.

vitabaks commented on automation/roles/patroni/tasks/main.yml:
```diff
@@ -942,8 +939,12 @@
-        - { name: '{{ patroni_superuser_username }}', password: '{{ patroni_superuser_password }}', flags: 'SUPERUSER' }
-        - { name: '{{ patroni_replication_username }}', password: '{{ patroni_replication_password }}', flags: 'LOGIN,REPLICATION' }
+        - { name: "{{ patroni_superuser_username }}", password: "{{ patroni_superuser_password }}", flags: "SUPERUSER" }
+        - {
+            name: "{{ patroni_replication_username }}",
+            password: "{{ patroni_replication_password }}",
+            flags: "LOGIN,REPLICATION",
+          }
```
```suggestion
        - { name: "{{ patroni_replication_username }}", password: "{{ patroni_replication_password }}", flags: "LOGIN,REPLICATION" }
```

I don't see the point in such formatting for just one line.

pat-s replied:
This is what prettier does. There are built-in rules for when there is a block-like style and for when not.

There is not really a way to have a custom view on these, either going with prettier or leaving it out entirely (your choice).

vitabaks replied:
I get it, thanks for clarifying.

vitabaks commented on automation/roles/pgbackrest/tasks/auto_conf.yml:
```diff
@@ -10,11 +9,26 @@
-        - { option: "repo1-s3-bucket", value: "{{ PGBACKREST_S3_BUCKET | default(aws_s3_bucket_name | default(patroni_cluster_name + '-backup')) }}" }
-        - { option: "repo1-s3-endpoint", value: "{{ PGBACKREST_S3_ENDPOINT | default('s3.' + (aws_s3_bucket_region | default(server_location)) + '.amazonaws.com') }}" }
-        - { option: "repo1-s3-region", value: "{{ PGBACKREST_S3_REGION | default(aws_s3_bucket_region | default(server_location)) }}" }
+        - {
+            option: "repo1-s3-key",
+            value: "{{ PGBACKREST_S3_KEY | default(lookup('ansible.builtin.env', 'AWS_ACCESS_KEY_ID')) }}",
+          }
```
It looks quite readable and compact in a single line. The lines are not long enough to justify such formatting.

This comment applies to the rest of the code lines.

pat-s replied:
See my previous comments about `prettier`.

vitabaks commented on automation/roles/pgbouncer/tasks/main.yml:
```diff
@@ -100,22 +100,21 @@
       when: not pgbouncer_bin.stat.exists
-  when: ansible_os_family == "RedHat" and
-        postgresql_packages|join(" ") is search("postgrespro")
+  when: ansible_os_family == "RedHat" and postgresql_packages|join(" ") is search("postgrespro")
```
```suggestion
  when:
    - ansible_os_family == "RedHat"
    - postgresql_packages|join(" ") is search("postgrespro")
```
we can use a list instead of `and`

vitabaks commented on automation/roles/pre-checks/tasks/huge_pages.yml:
```diff
@@ -99,7 +99,7 @@
 
     # if huge_pages_auto_conf is true
-    - name: "HugePages | Specify \"vm.nr_hugepages\" with value \"{{ huge_pages_required }}\" in sysctl_conf variable"
+    - name: 'HugePages | Specify "vm.nr_hugepages" with value "{{ huge_pages_required }}" in sysctl_conf variable'
```
```suggestion
    - name: "HugePages | Specify vm.nr_hugepages with value {{ huge_pages_required }} in sysctl_conf variable"
```

pat-s commented on automation/roles/pgbackrest/tasks/main.yml:
```diff
@@ -233,8 +234,7 @@
   when:
-    - (pgbackrest_repo_host is defined and pgbackrest_repo_host | length > 0) or
-      (pgbackrest_conf.global | selectattr('option', 'equalto', 'backup-standby') | map(attribute='value') | list | last | default('') == 'y')
+    - (pgbackrest_repo_host is defined and pgbackrest_repo_host | length > 0) or (pgbackrest_conf.global | selectattr('option', 'equalto', 'backup-standby') | map(attribute='value') | list | last | default('') == 'y')
```
No strong opinion. But this change might actually work together with `prettier`.

vitabaks commented:
> prettier -w .
> editorconfig-checker .

Perhaps we should automate the running of these commands in the [Makefile](https://github.com/vitabaks/autobase/blob/master/Makefile) and/or describe their use in [CONTRIBUTING.md](https://github.com/vitabaks/autobase/blob/master/CONTRIBUTING.md)

pat-s commented:
Usually both is good to have/needed to enforce them, yes. I do this in all of my projects. Though I don't use GH actions, hence I don't have workflows to c/p.

vitabaks commented:
@pat-s I changed the Python version from **3.13** to **3.12** for compatibility with **Ubuntu 24.04**, which I use in Gitpod and which is also the default distribution when deploying to cloud providers.

vitabaks commented on .config/.yamllint:
```diff
@@ -3,13 +3,18 @@ extends: default
 rules:
   line-length:
-    max: 160
+    max: 300
```
Why do we allow such long strings now?

If necessary, we can make an exception for individual files or lines, but not for the entire project.

pat-s replied:
Feel free to change and tweak the affected lines. (I just did this to have a clean run initially with cleanup work being done later). 

I personally don't care about line length as I am wrapping in my editors. Hence I usually disable this check entirely. 

But yes, 300 is quite ambiguous and if the check should stay, a lower value with proper adoption is surely better.

vitabaks commented on .config/.yamllint:
```diff
@@ -3,13 +3,18 @@ extends: default
 rules:
   line-length:
-    max: 160
+    max: 300
```
```suggestion
    max: 160
```
